### ISSUE TITLE
Git-Fixes

### ISSUE THREAD

**Issue by lapplislazuli** [Tritlo/Endemic#21]
It would be great if the fixes produced actually match the git .patch format so people can apply it easily, or add it to CI Tasks. 

- [x] Write patches to file(s)
- [x] git-ify patches 
- [ ] remove color?
- [ ] add metadata ? 

here is a [shorter example from SO](https://stackoverflow.com/questions/987372/what-is-the-format-of-a-patch-file) what the git patch means, and [here is a longer one](https://stackoverflow.com/questions/2529441/how-to-read-the-output-from-git-diff?noredirect=1&lq=1).

One example I have from another project / repository: 

```patch
From 62959418d5599f99da89be3fbc948660dd9e9489 Mon Sep 17 00:00:00 2001
From: lapplislazuli <Leonhard.Applis@Protonmail.com>
Date: Tue, 15 Jun 2021 08:23:34 +0200
Subject: [PATCH] test2

---
 .../test/java/elki/clustering/kmeans/ExponionKMeansTest.java  | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/elki-clustering/src/test/java/elki/clustering/kmeans/ExponionKMeansTest.java b/elki-clustering/src/test/java/elki/clustering/kmeans/ExponionKMeansTest.java
index 6fdd3c687..886d7132f 100644
--- a/elki-clustering/src/test/java/elki/clustering/kmeans/ExponionKMeansTest.java
+++ b/elki-clustering/src/test/java/elki/clustering/kmeans/ExponionKMeansTest.java
@@ -45,4 +45,15 @@ public class ExponionKMeansTest extends AbstractClusterAlgorithmTest {
     assertFMeasure(db, result, 0.998005);
     assertClusterSizes(result, new int[] { 199, 200, 200, 200, 201 });
   }
+
+    // Tests Issue 87
+    @Test
+    public void testKMeansExponion_SingleCluster_shouldPutAllInOneCluster() {
+        Database db = makeSimpleDatabase(UNITTEST + "different-densities-2d-no-noise.ascii", 1000);
+        Clustering<?> result = new ELKIBuilder<ExponionKMeans<DoubleVector>>(ExponionKMeans.class) //
+                .with(KMeans.K_ID, 1) //
+                .with(KMeans.SEED_ID, 7) //
+                .build().autorun(db);
+        assertClusterSizes(result, new int[] { 1000});
+    }
 }
-- 
2.14.2.windows.3

```
Later we can change the author to endemic, just for now I would keep it anonymus or something like "Repair Bot".

**Comment by lapplislazuli** [comment]
Writing to files is done with 87bbaca8b833ba5b333b0ef8f4b1694fe6fa1616 but has placeholders for configuration #18 , namely the base-folder where to write the patches and the locale of the time.

**Comment by Tritlo** [comment]
Yeah so these are the SO answers I based the `ppDiff` on actually. So it should work?

**Comment by Tritlo** [comment]
Let's leave the metadata for later, and leave it at just patches currently. We can dive into the metadata if we want to start making automatic pull requets or CI, but being able to just apply the files is enough for CI integration at least

**Comment by lapplislazuli** [comment]
Changes introduced in 0b5359f605be55f1fe03c43929d4dfa9b377d737, CI Failing due to outdated tests.

**Comment by lapplislazuli** [comment]
I agree that we should separate the meta-data stuff for further work. 
A git-patch bot or something would be feature creep, and is definitely a separate project. let's first make the library a hit 😄

**Comment by lapplislazuli** [comment]
Tests fixed in 95b226376bacf277e3a8818686d64f94f40fba6d
